### PR TITLE
Remove packet number re-randomization requirement

### LINKED ISSUE #823: Re-randomization for packet numbers
The draft is pretty inconsistent about what packet number to use in response to version negotiation or stateless retry.  Since no one implements re-randomization, just move to an increment-by-one.

### PR DESCRIPTION
**Pull request by martinthomson**
I don't believe that anyone actually implemented this, and it is
unnecessary.

Closes #823.

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -907,9 +896,9 @@ When the client receives a Version Negotiation packet from the server, it should
 select an acceptable protocol version.  If the server lists an acceptable
 version, the client selects that version and reattempts to create a connection
 using that version.  Though the contents of a packet might not change in
-response to version negotiation, a client MUST increase the packet number it
-uses on every packet it sends.  Packets MUST continue to use long headers and
-MUST include the new negotiated protocol version.
+response to version negotiation, a client MUST increase the packet number by one
```
Comment (janaiyengar) [inline draft-ietf-quic-transport.md]: This requirement of increment by 1 is broken around migration. This should probably make an exception for that, with a pointer to Section 7.6... but that's probably an overkill. How about "a client MUST increase the packet number by one for every packet it sends in response to version negotiation."

Comment (martinthomson) [reply draft-ietf-quic-transport.md]: Right, though you won't be migrating during version negotiation, so your fix works.

**Comment by janaiyengar** [comment]
This is fine, but I would just close it. We've agreed that we don't need randomization.

**Comment by martinthomson** [comment]
Since #850 isn't resolving, I'm going to merge this and then we can continue to fight about whether 0 is special or not at our leisure.

**Pull request merged**